tensorflow_model_optimization/python/core/clustering/keras/clustering_callbacks.py

```diff
@@ -0,0 +1,76 @@
+
+import tensorflow as tf
+
+from tensorflow import keras
+from tensorflow_model_optimization.python.core.keras import compat
+
+class ClusteringSummaries(keras.callbacks.TensorBoard):
+    """Helper class to create tensorboard summaries for the clustering progress.
+
+    This class is derived from tf.keras.callbacks.TensorBoard and just adds
+    functionality to write histogramms with batch-wise frequency.
+
+    Arguments:
+        log_dir: The path to the directory where the log files are saved
+        cluster_update_freq: determines the frequency of updates of the
+            clustering histograms. Same behaviour as parameter update_freq of
+            the base class, i.e. it accepts `'batch'`, `'epoch'` or integer.
+    """
+
+    def __init__(self,
+                 log_dir='logs',
+                 cluster_update_freq='epoch',
+                 **kwargs):
+        super(ClusteringSummaries, self).__init__(
```

Comment:
Should we validate log_dir here, similar to how it's done in PruningSummaries?

Reply:
We could, but I would leave that to the base class and/or the filewriter as they are actually using the `log_dir` directly. What do you think?

Reply:
I would agree, however, I would also try to make the behaviour of all APIs in MOT similar wherever possible. I guess we at least should make sure that passing invalid log_dir value would cause similar behaviour of __init__ in "summary" callback in both APIs. Do you know if tf.summary calls below raise an exception in such a case?

Reply:
Just as a pass-by, like Anton, I would also agree that the base class should be responsible. It happens to be that there is no error message today though and some sort of silent failure (I think maybe it just doesn't create a file - you can check), which a user ran into. Until we can make the base classes error well, this is useful.